### PR TITLE
Used IE_THROW macro

### PR DESCRIPTION
**Pull request by apankratovantonp**

### CODE REVIEW

File: inference-engine/include/ie_common.h [hunk 1]
```diff
@@ -378,26 +378,33 @@ struct ThrowNow final {
...
 /// @endcond
 
 /**
  * @def IE_THROW
- * @brief A macro used to throw specefied exception with a description
+ * @brief A macro used to throw specfied exception with a description
  */
-#define IE_THROW(ExceptionType)     \
-    InferenceEngine::details::ThrowNow<InferenceEngine::ExceptionType>{} <<= std::stringstream{} << IE_LOCATION
+#define IE_THROW(...) OV_PP_OVERLOAD(IE_THROW, __VA_ARGS__)
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_common.h:395]: `OV_PP_OVERLOAD` is not defined here..

File: inference-engine/include/ie_common.h [hunk 2]
```diff
@@ -378,26 +378,33 @@ struct ThrowNow final {
...
+
+#define IE_THROW_1(ExceptionType)                                                                                           \
+    InferenceEngine::details::ThrowNow<InferenceEngine::ExceptionType> {} <<= std::stringstream {}                          \
+    << IE_LOCATION << ' ' << InferenceEngine::details::ExceptionTraits<InferenceEngine::ExceptionType>::string() << ' '
 /// @endcond
 
 /**
  * @def IE_THROW
- * @brief A macro used to throw specefied exception with a description
+ * @brief A macro used to throw specfied exception with a description
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_common.h:393]: `specfied` -> `specified`

File: inference-engine/include/ie_common.h (hunk 1 above)
Comment (apankratovantonp) [reply inference-engine/include/ie_common.h:397]: fixed

File: inference-engine/include/ie_common.h [hunk 3]
```diff
@@ -20,6 +20,8 @@
 #include <stdexcept>
 #include <iterator>
 
+#include <openvino/pp.hpp>
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_common.h:23]: `pp` is not a part of OpenVINO release package. I suppose we cannot use it in public headers.

**Pull request merged**